src/clipping.jl

```diff
@@ -16,8 +16,36 @@ Clip the subject `geometry` with `other` geometry using clipping `method`.
 """
 function clip end
 
+function clip(poly::Polygon, other::Geometry, method::ClippingMethod)
+  b = boundary(other)
+  if b isa Multi
+    for r in parent(b)
+      poly = clip(poly, r, method)
+      if isnothing(poly)
+        return nothing
+      end
+    end
+    return poly
+  else
+    return clip(poly, b, method)
+  end
+end
+
+function clip(gs::Domain, other::Geometry, method::ClippingMethod)
+  polys = PolyArea[]
+  for i in 1:nelements(gs)
+    poly = clip(element(gs, i), other, method)
+    if !isnothing(poly)
+      push!(polys, poly)
+    end
+  end
+  n = length(polys)
+  return n == 0 ? nothing : (n == 1 ? polys[1] : GeometrySet(polys))
+end
```

Comment:
this could be written as

```julia
clipped = filter(!isnothing, [clip(geom, other, method) for geom in dom])
isempty(clipped) ? nothing : GeometrySet(clipped)
```

but I am starting to think that we should delay these higher-level methods to a separate PR.  We can even consider a new `Clip` transform that handles these higher-level calls.

Reply:
Applied the suggestion, function moved to weileratherton.jl.